rules/the-main-principles-of-clean-architecture/rule.md

```diff
@@ -8,24 +8,92 @@ created: 2019-04-14T21:34:38.0000000Z
 authors:
 - title: Jason Taylor
   url: https://ssw.com.au/people/jason-taylor
+- title: Luke Parker
+  url: https://ssw.com.au/people/luke-parker
 related: []
 redirects:
 - do-you-know-the-main-principles-of-clean-architecture
 
 ---
 
-With Clean Architecture, the Domain and Application layers are at the centre of the design. This is known as the Core of the application. The Domain layer contains the enterprise logic and types, and the Application layer contains the business logic and types. The difference being that enterprise logic could be shared with other systems whereas business logic would typically be specific to this system.
+Clean Architecture at its core is about creating a Separation of Concerns, and building it in such a way that it becomes hard to violate this core principal. The outcome of this is a design that is independent of frameworks, UIs, databases and anything external. This makes it easy to test, swap out frameworks, UIs, and databases, and it makes it easy to change the business logic without impacting the rest of the system.
 
-![Figure: Onion View of Clean Architecture](ca-diagram.png)  
+![Figure: Onion View of Clean Architecture](ca-diagram.png)
 
 <!--endintro-->
 
-Instead of having Core depend on data access and other infrastructure concerns, we invert these dependencies, therefore Infrastructure and Presentation depend on Core. This is achieved by adding abstractions, such as interfaces or abstract base classes, to the Application layer. Layers outside of Core, such as Infrastructure and Persistence, then implement these abstractions.
+There are 4 layers to Clean Architecture, and each layer has a specific purpose. The 4 layers are:
 
-A good example is the implementation of the Repository pattern. Within this design, we would first add an IRepository interface to the Application layer. Next, we would implement this interface within Persistence by creating a Repository class using our preferred data access technology. Finally, within Core the logic we write will only use the IRepository interface, so Core will remain independent of data access concerns.
+## Layers
+
+### Domain
+
+The Domain layer contains the enterprise logic and types. This is the core of the application and contains the business logic. This layer should not depend on anything outside of itself. This layer typically defines the models and data structures that represent the business entities and concepts.
+
+Examples:
+
+ - Entities
+ - Value Objects
+ - Domain Events
+
+### Application
+
+The Application layer contains the business logic and types. This layer is dependent on the Domain layer, but not on anything outside of itself. This layer typically defines the application services that implement the use cases of the system. These services orchestrate the flow of data using the domain entities and types.
```

Comment:
> Instead of having Core (Domain & Application layers) ...

I have added this back, with the additional expansion of what Core is. Are you happy it is resolved now?